docs/contribute/plugin/develop_plugin_c.md

```diff
@@ -4,199 +4,238 @@ sidebar_position: 2
 
 # Develop WasmEdge Plug-in in C API
 
-WasmEdge provides a C++ based API for registering extension modules and host functions. While the WasmEdge language SDKs allow registering host functions from a host (wrapping) application, the plugin API allows such extensions to be incorporated into WasmEdge's own building and releasing process.
-
-## Prerequisites
-
-For developing the WasmEdge plug-in in C API, please [install WasmEdge](../../start/install.md#install) first.
+By developing a plug-in, one can extend the functionality of WasmEdge and customize it to suit specific needs. WasmEdge provides a C++ based API for registering extension modules and host functions. While the WasmEdge language SDKs allow registering host functions from a host (wrapping) application, the plug-in API allows such extensions to be incorporated into WasmEdge's building and releasing process. Here is a flowchart showing all the steps needed for developing WasmEdge Plug-in -
+
+```mermaid
+graph LR;
+A[Developing WasmEdge Plug-in in C]
+A --> B(Set up the development environment)
+A --> C(Create a WasmEdge plug-in project)
+A --> D(Write the plugin code)
+A --> E(Build the plugin)
+A --> F(Test and debug the plug-in)
+B --> E
+C --> D
+D --> E
+```
 
-## Example
+This flowchart illustrates developing a WasmEdge plug-in, showcasing the steps from choosing a programming language to finalizing and releasing the plug-in.
 
-Assume that the plug-in example is in the file `testplugin.c`.
+## Set up the development environment
 
-### Host Functions
+To start developing WasmEdge plug-ins, it is essential to correctly set up the development environment. This section provides step-by-step instructions for WasmEdge plug-in development -
 
-The goal of the plug-in is to provide the host functions which can be imported when instantiating WASM.
+- **Install a WasmEdge runtime**: You can download the latest version of WasmEdge from [GitHub repository](https://github.com/wasmEdge/wasmEdge). Here are the installation instructions for a different operating system for [installing WasmEdge](/develop/build-and-run/install.md).
```

Comment:
You can briefly provide the link to `../../start/install.md`.
And also, please guide the developers to set up the build environment on Linux or other platforms: `../source/os/linux.md`